test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -90,24 +92,33 @@ AndroidArgs
     }
 }
 
+// Changes these based on type
 val AndroidArgs.isDontAutograntPermissions
     get() = !(grantPermissions.isNotNull() && (grantPermissions.equals("all")))
 
 val AndroidArgs.isInstrumentationTest
-    get() = appApk.isNotNull() &&
-            testApk.isNotNull() ||
-            additionalAppTestApks.isNotEmpty() &&
-            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() })
+    get() = if (type != null) (type == Type.INSTRUMENTATION) else validateInstrumentation()
+
+fun AndroidArgs.validateInstrumentation() = (appApk.isNotNull() && testApk.isNotNull() ||
+        additionalAppTestApks.isNotEmpty() &&
+        (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
 
 val AndroidArgs.isRoboTest
+    get() = if (type != null) type == Type.ROBO else validateRobo()
+
+fun AndroidArgs.validateRobo() = (appApk.isNotNull() && (roboDirectives.isNotEmpty() || roboScript.isNotNull()))
+
+val AndroidArgs.checkForSanityRobo
     get() = appApk.isNotNull() &&
-            (roboDirectives.isNotEmpty() || roboScript.isNotNull())
+        testApk.isNull() &&
+        roboScript.isNull() &&
+        additionalAppTestApks.isEmpty()
 
 val AndroidArgs.isSanityRobo
-    get() = appApk.isNotNull() &&
-            testApk.isNull() &&
-            roboScript.isNull() &&
-            additionalAppTestApks.isEmpty()
+    get() = if (type != null) (type == Type.ROBO) && checkForSanityRobo else checkForSanityRobo
+
+val AndroidArgs.isGameLoop
+    get() = if (type == null) false else (type == Type.GAMELOOP)
```

Comment:
I think we could simplify it:
```suggestion
    get() = type == Type.GAMELOOP
```
I think enums have `==` implemented as safe calls out of the box

Reply:
I will leave this one as is.